### PR TITLE
Clipboard

### PR DESCRIPTION
**Pull request by munkk**

### CODE REVIEW

File: src/core/clipboard/clipboard.view.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        const { shortcut }  = clipboardState;
+
+        const commands = {
+            copy: new Command({
+                source: 'clipboard.view',
+                canExecute: () => selectionState.items.length > 0,
+                execute: () => {
+                    const selectionService = new SelectionService(model);
+                    const rowSelector = new RowSelector(model);
+                    const { items } = selectionState;
```
Comment (klumba12) [inline src/core/clipboard/clipboard.view.js]: try to make it like { items } = model.selection(), do not close model state please, there is no guarantee that state will be in a valid position

File: src/core/clipboard/clipboard.view.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+import { Disposable } from '../infrastructure/disposable'
+
+export class ClipboardView {
+    constructor(model, table, commandManager) {
+        super();
+
+        this.model = model;
+        this.table = table;
+
+        const selectionCommandManager = new SelectionCommandManager(model, commandManager);
```
Comment (klumba12) [inline src/core/clipboard/clipboard.view.js]: commandManager is enought

File: src/core/clipboard/clipboard.view.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        const action = model.action().shortcut;
```
Comment (klumba12) [inline src/core/clipboard/clipboard.view.js]: why action? please call it shortcut

File: src/core/clipboard/clipboard.view.js [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    }
+
+    onPaste(e) {
+        e.stopPropagation();
+        e.preventDefault();
+
+        const model = this.model;
+        const table = this.table;
+
+        const navigation = model.navigation();
```
Comment (klumba12) [inline src/core/clipboard/clipboard.view.js]: navigationState

File: src/lib/main/grid/grid.component.ts [hunk 5]
```diff
@@ -169,6 +169,7 @@ export class GridComponent extends RootComponent implements OnInit {
 
 		const listener = new EventListener(element, new EventManager(this));
 		const docListener = new EventListener(this.document, new EventManager(this));
+		docListener.on('paste', (e: ClipboardEvent) => model.clipboard({clipboardEvent: e}));
```
[the remark this answers is not in the document]
Comment (klumba12) [reply src/lib/main/grid/grid.component.ts]: this docListener okey, just don't push clipboardEvent

[the remark this answers is not in the document]
Comment (munkk) [reply src/lib/main/grid/grid.component.ts]: then how can I retrieve data from the event?

**Pull request closed without merging**